src/rest_vol_attr.c

```diff
@@ -1271,7 +1375,115 @@ RV_attr_get(void *obj, H5VL_attr_get_args_t *args, hid_t dxpl_id, void **req)
+
+                    strcpy(host_header, host_string);
+
+                    curl_headers = curl_slist_append(
+                        curl_headers, strncat(host_header, loc_obj->domain->u.file.filepath_name,
+                                              host_header_len - strlen(host_string) - 1));
+
+                    /* Disable use of Expect: 100 Continue HTTP response */
+                    curl_headers = curl_slist_append(curl_headers, "Expect:");
+
+                    /* Redirect cURL from the base URL to
+                     * "/groups/<id>/attributes/<attr name>",
+                     * "/datatypes/<id>/attributes/<attr name>"
+                     * or
+                     * "/datasets/<id>/attributes/<attr name>",
+                     * depending on the type of the object the attribute is attached to. */
+                    if (RV_set_object_type_header(parent_obj_type, &parent_obj_type_header) < 0)
+                        FUNC_GOTO_ERROR(H5E_ATTR, H5E_BADVALUE, FAIL,
+                                        "can't get path header from parent object type");
+
+                    if ((url_len =
+                             snprintf(request_url, URL_MAX_LENGTH, "%s/%s/%s?%s&include_attrs=1", base_URL,
+                                      parent_obj_type_header, parent_obj_URI, request_idx_type)) < 0)
+                        FUNC_GOTO_ERROR(H5E_ATTR, H5E_SYSERRSTR, FAIL, "snprintf error");
+
+                    if (url_len >= URL_MAX_LENGTH)
+                        FUNC_GOTO_ERROR(H5E_ATTR, H5E_SYSERRSTR, FAIL,
+                                        "attribute open URL exceeded maximum URL size");
+
+                    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPHEADER, curl_headers))
+                        FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTSET, FAIL, "can't set cURL HTTP headers: %s",
+                                        curl_err_buf);
+                    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPGET, 1))
+                        FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTSET, FAIL,
+                                        "can't set up cURL to make HTTP GET request: %s", curl_err_buf);
+                    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_URL, request_url))
+                        FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTSET, FAIL, "can't set cURL request URL: %s",
+                                        curl_err_buf);
+
+                    CURL_PERFORM(curl, H5E_ATTR, H5E_CANTGET, FAIL);
+
+                    if (0 > RV_parse_response(response_buffer.buffer, &loc_params->loc_data.loc_by_idx,
+                                              &found_attr_name, RV_copy_attribute_name_by_index))
+                        FUNC_GOTO_ERROR(H5E_ATTR, H5E_PARSEERROR, FAIL, "failed to retrieve attribute names");
+
+                    if (host_header)
+                        RV_free(host_header);
+                    if (url_encoded_attr_name)
+                        curl_free(url_encoded_attr_name);
```

Comment:
`host_header` and `url_encoded_attr_name` should be set to NULL so they aren't double-freed